src/circuit/transport.js

```diff
@@ -89,7 +101,7 @@ class Circuit {
         remoteAddr,
         localAddr
       })
-      const type = CircuitPB.Type === CircuitPB.Type.HOP ? 'relay' : 'inbound'
+      const type = request.Type === CircuitPB.Type.HOP ? 'relay' : 'inbound'
```

Comment:
I believe it should be following instead:

```suggestion
      const type = request.type === CircuitPB.Type.HOP ? 'relay' : 'inbound'
```

Reply:
And a reason TS failed to catch this is because `StreamHandler` is generic and on line 56 `streamHandler` was inferred as `StreamHandler<any>`.

Reply:
good catch! I also added a type for line 56: `/** @type {import('./circuit/stream-handler')<CircuitRequest>} */`